extension/dynamicconfig/model/collection_period_test.go

```diff
@@ -0,0 +1,52 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package model
+
+import (
+	"bytes"
+	"testing"
+)
+
+func TestCollectionPeriodProto(t *testing.T) {
+	var period CollectionPeriod = "MIN_5"
+	p := period.Proto()
+
+	if int(p) != 300 {
+		t.Errorf("improper conversion to proto")
+	}
+}
+
+func TestCollectionPeriodNotFoundProto(t *testing.T) {
+	var period CollectionPeriod = "DAY_69105"
+	p := period.Proto()
+
+	if int(p) != 0 {
+		t.Errorf("improper conversion to proto")
+	}
+}
+
+func TestCollectionPeriodHash(t *testing.T) {
```

Comment:
Great question, the string names map 1:1 to the enum names specified in the protobuf. Therefore "HR_1" would yield 3600, but "SEC_3600" would map to "0" (for the default "NONE" type in the enum).

In this case, the hashing algorithm takes only the numeric value of the field, not the string value. Hence "SEC_3600" and "SEC_60" would both be assigned the same hash, since both map to "0." However "HR_1" and "MIN_1" would be assigned to unique hashes. This setup prevents updates from occurring while the fundamental value of the period remains the same (i.e. 0).